### PR TITLE
Early stopping

### PR DESCRIPTION
**Pull request by kasra-hosseini**
Two parameters:
* **early_stopping_patience:** wait these many epochs, if the validation loss does not improve, stop OR False (deactivate early stopping)
~* **early_stopping_delta:** minimum change in the validation loss to be considered as improvement (normally zero)~

### CODE REVIEW

**Comment by kasra-hosseini** [comment]
@fedenanni Sorry, I can only add one reviewer. This is a small PR in which we add a simple "early stopping" functionality.

**Comment by kasra-hosseini** [comment]
Hi @mcollardanuy and @fedenanni , hopefully, this [commit](https://github.com/Living-with-machines/DeezyMatch/pull/74/commits/5ca48f1c642bf877a8882d1606cc3ad4081a051f) clarifies the changes in this PR. Would you please take a look and let me know if I should change anything?

File: inputs/input_dfm.yaml [hunk 1]
```diff
@@ -69,6 +69,11 @@ gru_lstm:
   # shuffle when creating DataLoader
   dl_shuffle: True 
   random_seed: 123
+  # Early stopping:
+  # Number of epochs with no improvement after which training will be stopped and
+  #   the model with the least validation loss will be saved
+  # If False, 0 or negative, early stopping will be deactivated
+  early_stopping_patience: False
```
Comment (fedenanni) [inline inputs/input_dfm.yaml:76]: It's a tiny thing, but maybe we could say that `early_stopping_patience` cannot be `True` at the moment (as far as I have seen), but needs to be a number. As by default is `False` people might think that they could make it `True` and we will use a "default" patience - which we could do in theory, but I think we don't. But I might have missed something.

Comment (kasra-hosseini) [reply inputs/input_dfm.yaml]: Good point, I removed `False`, see: https://github.com/Living-with-machines/DeezyMatch/pull/74/commits/5d2f16e7e5e3be8c2a39b2c247d645bbb920ced2

Comment (kasra-hosseini) [reply inputs/input_dfm.yaml]: Is this OK?

Comment (fedenanni) [reply inputs/input_dfm.yaml]: perfect

**Comment by fedenanni** [comment]
All good from my side

**Comment by mcollardanuy** [comment]
Hi @kasra-hosseini, I just tested it, all looks good!
